db/20180329014741_add_devise_to_users.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+class AddDeviseToUsers < ActiveRecord::Migration
+  def self.up
+    change_column_null :users, :email, false
+
+    change_table :users do |t|
+      ## Database authenticatable
+      t.string :encrypted_password, null: false, default: ""
+
+      ## Recoverable
+      t.string   :reset_password_token
+      t.datetime :reset_password_sent_at
+
+      ## Rememberable
+      t.datetime :remember_created_at
+
+      ## Trackable
+      t.integer  :sign_in_count, default: 0, null: false
+      t.datetime :current_sign_in_at
+      t.datetime :last_sign_in_at
+      t.string   :current_sign_in_ip
+      t.string   :last_sign_in_ip
+
+      ## Confirmable
+      # t.string   :confirmation_token
```

Comment:
are these just less necessary or more of a pain to implement?

Reply:
This migration was auto-generated by devise and I just stuck with the default things (this was commented out by default). But we can always look more into what these things do and see if we need them.

Reply:
oh cool, default settings is definitely good for now